common/app/model/dotcomrendering/DotcomRenderingDataModel.scala

```diff
@@ -228,10 +228,18 @@ object DotcomRenderingDataModel {
     })
 
     val bodyBlocks = DotcomRenderingUtils.blocksForLiveblogPage(page, blocks)
-    val keyEvents =
+
+    val allKeyEvents =
       blocks.requestedBodyBlocks
-        .flatMap(blocks => blocks.get("body:key-events"))
-        .getOrElse(Seq.empty[APIBlock])
+        .flatMap(bodyBlocks => bodyBlocks.get("body:key-events"))
+        .getOrElse(blocks.body.fold(Seq.empty[APIBlock])(_.filter(_.attributes.keyEvent.contains(true))))
```

Comment:
if this is just picking key events from the list of body blocks we can probably avoid using fold, as it's a bit difficult to read
```suggestion
        .getOrElse(blocks.body.filter(_.attributes.keyEvent.contains(true))))
```

Reply:
I think we started by using `.filter`, but we then had to do an additional `.getOrElse(Seq.empty[APIBlock])`, so we changed to a `.fold`, I'm open to either but neither seemed quite right!

Reply:
I was having a look at tat as well and I find the fold easier to read and understand, but I know those things are personal preferences :)